src/analysis/processing/qgsalgorithmcalculateoverlaps.cpp

```diff
@@ -52,6 +52,11 @@ void QgsCalculateVectorOverlapsAlgorithm::initAlgorithm( const QVariantMap & )
   addParameter( new QgsProcessingParameterFeatureSource( QStringLiteral( "INPUT" ), QObject::tr( "Input layer" ), QList< int >() << QgsProcessing::TypeVectorPolygon ) );
   addParameter( new QgsProcessingParameterMultipleLayers( QStringLiteral( "LAYERS" ), QObject::tr( "Overlay layers" ), QgsProcessing::TypeVectorPolygon ) );
   addParameter( new QgsProcessingParameterFeatureSink( QStringLiteral( "OUTPUT" ), QObject::tr( "Overlap" ) ) );
+
+  std::unique_ptr< QgsProcessingParameterNumber > gridSize = std::make_unique< QgsProcessingParameterNumber >( QStringLiteral( "GRIDSIZE" ),
```

Comment:
See how the QgsLayoutAtlasToImageAlgorithm handles the "DPI" parameter -- we should do that here. It defaults to "not set" and offers the gui option to clear the value back to not set. That's cleaner than a non-self-explanatory "-1" default value.

(That'll also make it cleaner when we add a default grid size option to the processing context itself.)